### PR TITLE
feat: server side HTML detection

### LINKED ISSUE #1126: Sanitization on Template Upload
With Templates (Form Configs) no longer being created by a trusted source we need to increase our efforts to ensure that 'dirty' inputs do not pollute our application.
On a Template creation or update the various inputs need to be sanitized to ensure only clean values are stored in the database.

This issue is specifically for the `/api/templates` path.
Potential package recommended by OWASP to help with sanitization :https://github.com/cure53/DOMPurify
An example on how we can wrap it to be isomorphic can be found here: https://github.com/kkomelin/isomorphic-dompurify/blob/master/src/index.js

### PR DESCRIPTION
**Pull request by Moro-Code**
# Summary | Résumé

closes #1126 

This PR adds HTML detection in the form template JSON when passed to the template upload/update API. 

If HTML is detected a 400 response is returned with an error informing the requester that HTML has been detected 

# Test instructions | Instructions pour tester la modification

Look at test cases and see if there are ways around it. My test case is currently having html in the definition of a dynamic row since thats the most nested.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

N/A

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: lib/middleware/jsonValidator.ts [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+    // we want to ensure that the only thing in the document is a text element
+    const parsedString = htmlparser2.parseDocument(value);
+
+    // if there are zero children its an empty string, if there is more than one child there is html
+    if (parsedString.children.length === 0) return;
+    else if (parsedString.children.length !== 1) throw new Error(`HTML detected in JSON`);
+
+    const element = parsedString.children[0];
+    // if the parsed element is not of type text its html
+    if (key === "subElements") {
```
Comment (craigzour) [inline lib/middleware/jsonValidator.ts:36]: Is not that some debugging code?

File: lib/middleware/jsonValidator.ts [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+/**
+ * This function when passed into the jsonschema validate function
+ * will receive all possible key value pairs defined in the schema
+ * and will check string values to ensure that there is not HTML
+ * @param object
+ * @param key
+ */
+const htmlChecker: PreValidatePropertyFunction = (object, key) => {
+  const value = object[key];
+  if (key === "subElements") {
```
Comment (craigzour) [inline lib/middleware/jsonValidator.ts:21]: Is that supposed to help debug the validation in a dev environment?

File: lib/middleware/jsonValidator.ts (hunk 1 above)
Comment (Moro-Code) [reply lib/middleware/jsonValidator.ts]: Yes good catch I thought I had removed those

**Review by craigzour — APPROVED** [review]
LGTM :)

**Pull request merged**